python/dgl/data/chem/utils.py

```diff
@@ -10,42 +12,583 @@
+        Chiral tags to consider. Default: ``rdkit.Chem.rdchem.ChiralType.CHI_UNSPECIFIED``,
+        ``rdkit.Chem.rdchem.ChiralType.CHI_TETRAHEDRAL_CW``,
+        ``rdkit.Chem.rdchem.ChiralType.CHI_TETRAHEDRAL_CCW``,
+        ``rdkit.Chem.rdchem.ChiralType.CHI_OTHER``.
+    """
+    if allowable_set is None:
+        allowable_set = [Chem.rdchem.ChiralType.CHI_UNSPECIFIED,
+                         Chem.rdchem.ChiralType.CHI_TETRAHEDRAL_CW,
+                         Chem.rdchem.ChiralType.CHI_TETRAHEDRAL_CCW,
+                         Chem.rdchem.ChiralType.CHI_OTHER]
+    return one_hot_encoding(atom.GetChiralTag(), allowable_set, encode_unknown)
+
+
+def atom_mass(atom, coef=0.01):
+    """Get the mass of an atom and scale it.
+
+    Parameters
+    ----------
+    atom : rdkit.Chem.rdchem.Atom
+        RDKit atom instance.
+    coef : float
+        The mass will be multiplied by ``coef``.
+
+    Returns
+    -------
+    list
+        List containing one float only.
+    """
+    return [atom.GetMass() * coef]
+
+
+class ConcatFeaturizer(object):
+    """Concatenate the evaluation results of multiple functions as a single feature.
+
+    Parameters
+    ----------
+    data_type
+        The type of objects that we want to featurize. E.g. ``rdkit.Chem.rdchem.Atom``
+        or ``rdkit.Chem.rdchem.Bond``.
+    func_list : list
+        List of functions for computing molecular descriptors from objects of type
+        ``data_type``. Each function is of signature
+        ``func(data_type) -> list of float or bool or int``. The resulting order of
+        the features will follow that of the functions in the list.
+    """
+
+    def __init__(self, data_type, func_list):
+        self.data_type = data_type
+        self.func_list = func_list
+
+    def __call__(self, x):
+        """Featurize the input data.
+
+        Parameters
+        ----------
+        x : self.data_type
+            Data to featurize.
+
+        Returns
+        -------
+        list
+            List of feature values, which can be of type bool, float or int.
+        """
+        assert isinstance(x, self.data_type), \
+            'Expect the input data to be of type {}, got {}'.format(
+                self.data_type, type(x))
+        return list(itertools.chain.from_iterable(
+            [func(x) for func in self.func_list]))
+
+
 class BaseAtomFeaturizer(object):
-    """An abstract class for atom featurizers
+    """An abstract class for atom featurizers.
 
-    All atom featurizers that map a molecule to atom features should subclass it.
-    All subclasses should overwrite ``_featurize_atom``, which featurizes a single
-    atom and ``__call__``, which featurizes all atoms in a molecule.
+    Loop over all atoms in a molecule and featurize them with the ``featurizer_funcs``.
+
+    **We assume the resulting DGLGraph will not contain any virtual nodes.**
+
+    Parameters
+    ----------
+    featurizer_funcs : dict
```

Comment:
Why does this need to be a dict?

Reply:
I think this is the easiest thing we can do when we want to construct multiple features at one pass.

Reply:
I don't like this but it's fine. Could you add an example at the doc?

Reply:
Done.